src/components/generics/generic-dialog.tsx

```diff
@@ -0,0 +1,107 @@
+import CloseIcon from "@mui/icons-material/Close";
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  IconButton
+} from "@mui/material";
+import { ReactNode } from "react";
+
+/**
+ * Component properties
+ */
+interface Props {
+  title: string;
+  confirmButtonText?: string;
+  cancelButtonText?: string;
+  onClose: () => void;
+  onCancel: () => void;
+  onConfirm?: () => void | Promise<void> | unknown;
+  open: boolean;
+  error?: boolean;
+  fullScreen?: boolean;
+  fullWidth?: boolean;
+  disableEnforceFocus?: boolean;
+  disabled?: boolean;
+  ignoreOutsideClicks?: boolean;
+  children: ReactNode;
+  maxWidth?: "xs" | "sm" | "md" | "lg" | "xl";
+}
+
+/**
+ * Generic dialog component
+ *
+ * @param props component properties
+ */
+const GenericDialog = ({
+  open,
+  confirmButtonText,
+  cancelButtonText,
+  onClose,
+  onCancel,
+  title,
+  onConfirm,
+  error,
+  fullScreen,
+  fullWidth,
+  maxWidth,
+  disableEnforceFocus,
+  disabled,
+  ignoreOutsideClicks,
+  children
+}: Props) => {
```

Comment:
Dialog component has a list of props, some which are unused. For me this is fine, but if you want to more concise code then maybe the unnecessary props could be dropped for now.

Reply:
I think these are useful options to have in the generic dialog.